test/c/test_target.c

```diff
@@ -0,0 +1,303 @@
+    int result = Ok;
+    // Add an X Gate.
+
+    // This operation is global, no property map is provided
+    qk_target_add_instruction(target, qk_target_entry_new(QkGate_X));
+
+    // Number of qubits of the target should not change.
+    size_t current_num_qubits = qk_target_num_qubits(target);
+    if (current_num_qubits != 1) {
+        printf("The number of qubits this target is compatible with is not 1: %zu",
+               current_num_qubits);
+        return EqualityError;
+    }
+    size_t current_size = qk_target_num_instructions(target);
+    if (current_num_qubits != 1) {
+        printf("The size of this target is not correct: Expected 1, got %zu", current_size);
+        return EqualityError;
+    }
+
+    // Add a CX Gate.
+    // Create prop_map for the instruction
+    // Add property for (0, 1)
+    QkTargetEntry *cx_entry = qk_target_entry_new(QkGate_CX);
+    uint32_t qargs[2] = {0, 1};
+    double inst_error = 0.0090393;
+    double inst_duration = 0.020039;
+    qk_target_entry_add_property(cx_entry, qargs, 2, inst_duration, inst_error);
+
+    qk_target_add_instruction(target, cx_entry);
+
+    // Number of qubits of the target should change to 2.
+    current_num_qubits = qk_target_num_qubits(target);
+    if (current_num_qubits != 2) {
+        printf("The number of qubits this target is compatible with is not 2: %zu",
+               current_num_qubits);
+        result = EqualityError;
+        goto cleanup;
+    }
+    current_size = qk_target_num_instructions(target);
+    if (current_num_qubits != 2) {
+        printf("The size of this target is not correct: Expected 2, got %zu", current_size);
+        return EqualityError;
+    }
+
+    // Add a CRX Gate.
+    // Create prop_map for the instruction
+    // Add property for (0, 1)
+    double crx_params[1] = {3.14};
+    QkTargetEntry *crx_entry = qk_target_entry_new_fixed(QkGate_CRX, crx_params);
+    uint32_t crx_qargs[2] = {1, 2};
+    double crx_inst_error = 0.0129023;
+    double crx_inst_duration = 0.92939;
+    qk_target_entry_add_property(crx_entry, crx_qargs, 2, crx_inst_duration, crx_inst_error);
+    // CX Gate is not paramtric.
+
+    qk_target_add_instruction(target, crx_entry);
+
+    // Number of qubits of the target should change to 3.
+    current_num_qubits = qk_target_num_qubits(target);
+    if (current_num_qubits != 3) {
+        printf("The number of qubits this target is compatible with is not 3: %ld",
+               current_num_qubits);
+        result = EqualityError;
+        goto cleanup;
+    }
+    current_size = qk_target_num_instructions(target);
+    if (current_num_qubits != 3) {
+        printf("The size of this target is not correct: Expected 3, got %zu", current_size);
+        return EqualityError;
+    }
+
+cleanup:
+    qk_target_free(target);
+    return result;
+}
+
+/**
+ * Test updating an instruction property in the Target using
+ * `update_instruction_property`.
+ */
+int test_target_update_instruction(void) {
```

Comment:
This is already nice but it would be good to also check the other error codes are raised as expected 🙂

Reply:
Hm I don't see `QkExitCode_TargetQargMismatch` or `QkExitCode_TargetInstAlreadyExists` being used anywhere, but maybe I'm missing something 🤔

Reply:
Ok sorry my initial comment was not clear enough: I meant it would be good to have tests for all types of ExitCodes we have defined 🙂